### PR TITLE
Add dependency version management using dep

### PR DESCRIPTION
**Pull request by VoyTechnology**
The change would allow to have better control over the dependencies and
their changing versions.

### CODE REVIEW

**Comment by VoyTechnology** [comment]
I am aware the CI has failed and I'm looking into it.

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  branch = "master"
+  name = "github.com/dustin/go-humanize"
+
+[[constraint]]
+  name = "github.com/hashicorp/terraform"
+  version = "0.11.7"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/pyToshka/go-virtualbox"
```
Comment (ringods) [inline Gopkg.toml]: Could you update this dependency to the root repo? The `pyToshka` version is a fork. I have pulled in all the changes from this fork in the root repo:

root repo = `github.com/terra-farm/go-virtualbox`

Comment (VoyTechnology) [reply Gopkg.toml]: It was generated by `dep init` based on the code in the repo at the time, specifically:

https://github.com/terra-farm/terraform-provider-virtualbox/blob/master/provider/resource_vm.go#L17

If someone wants to file a PR to change the package (or I can do it), I will update the PR.

File: Gopkg.lock [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+    "html",
+    "html/atom",
+    "http2",
+    "http2/hpack",
+    "idna",
+    "internal/timeseries",
+    "lex/httplex",
+    "trace"
+  ]
+  revision = "f2499483f923065a842d38eb4c7f1927e6fc6e6d"
```
Comment (ringods) [inline Gopkg.lock]: @VoyTechnology this revision of `net/http2` does not contain the function `SetReuseFrames`. Can you update this revision manually to `ffcf1bedda3b04ebb15a168a59800a73d6dc0f4d` and try again?

I got this revision from another project on Github using `net/http2` with the function `SetReuseFrames`
https://github.com/SolarisYan/grpc-hello-with-gateway/blob/master/vendor/manifest

File: Gopkg.toml (hunk 1 above)
Comment (ringods) [reply Gopkg.toml]: I will leave it for now. Let's get the build working so that this can be merged.

**Comment by VoyTechnology** [comment]
My last commit fixed the issue. However `dep ensure` took 281.90s, which is less than ideal. Should I add caching in this PR? `dep ensure` will make sure the dependencies are of correct version, so there is no need to worry about old packages being used.

**Comment by ringods** [comment]
@VoyTechnology I merged this PR for now. If you still want to add caching, feel free to create a new PR.  Thanks already for providing the `dep` support.

**Pull request merged**